### ISSUE TITLE
ClassCastException for NRA instances 

### ISSUE THREAD

**Issue by rainoftime** [ultimate-pa/smtinterpol#68]
Related and fixed https://github.com/ultimate-pa/smtinterpol/issues/51

----------

Hi, for the following formulas

[cast1.txt](https://github.com/ultimate-pa/smtinterpol/files/4457609/cast1.txt)
[cast2.txt](https://github.com/ultimate-pa/smtinterpol/files/4457611/cast2.txt)

smtinterpol 5171d77796 throws  a ClassCastException
~~~~
Unexpected Exception: java.lang.ClassCastException: de.uni_freiburg.informatik.ultimate.smtinterpol.theory.cclosure.CCBaseTerm cannot be cast to de.uni_freiburg.informatik.ultimate.smtinterpol.theory.cclosure.CCAppTerm
(error "java.lang.ClassCastException: de.uni_freiburg.informatik.ultimate.smtinterpol.theory.cclosure.CCBaseTerm cannot be cast to de.uni_freiburg.informatik.ultimate.smtinterpol.theory.cclosure.CCAppTerm")
~~~~

**Comment by jhoenicke** [comment]
I guess this is resolved now, right?

**Comment by tanjaschindler** [comment]
It should be resolved now with 52389f7.
